src/main/kotlin/me/partlysanestudios/partlysaneskies/features/debug/PercyMode.kt

```diff
@@ -0,0 +1,154 @@
+
+                entity.writeToNBT(nbt)
+
+                jsonObject.add("entityData", Gson().toJsonTree(nbt))
+
+                jsonObject.add("class", Gson().toJsonTree(entity.javaClass.name))
+            } catch (exception: Exception) {
+                sendClientMessage("Error getting data for entity $name")
+                jsonArray.add(jsonObject)
+            }
+
+            jsonArray.add(jsonObject)
+        }
+
+        return jsonArray
+    }
+
+    private fun getInventoryDump(): JsonObject {
+        val nbt = NBTTagList()
+        PartlySaneSkies.minecraft.thePlayer?.inventory?.writeToNBT(nbt)
+        return Gson().toJsonTree(nbt)?.asJsonObject?: JsonObject()
+    }
+
+    private fun getPlayerDump(): JsonObject {
+        val nbt = NBTTagCompound()
+        PartlySaneSkies.minecraft.thePlayer?.writeToNBT(nbt)
+        return Gson().toJsonTree(nbt)?.asJsonObject?: JsonObject()
+    }
+
+    private fun getTimeDump(): JsonObject {
+        val fullObject = JsonObject()
+        // Format the Instant to a human-readable date and time
+        // Convert epoch time to LocalDateTime
+        val dateTime = LocalDateTime.ofInstant(Instant.now(), ZoneId.systemDefault())
+
+        // Format the LocalDateTime to a human-readable date and time
+        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
+        val formattedDate = dateTime.format(formatter)
+
+        fullObject.add("unix", JsonPrimitive(PartlySaneSkies.time))
+        formatter.withZone(ZoneId.systemDefault())
+        fullObject.add("formattedUtc", JsonPrimitive(formattedDate))
+
+        return fullObject
+    }
+
+    private fun getFullDump(): JsonObject {
+        val fullObject = JsonObject()
+
+        fullObject.add("time", getTimeDump())
+        if (config.debugCurrentScreenDump) fullObject.add("currentScreen", getCurrentScreenDump())
+        if (config.debugEntityDump) fullObject.add("entities", getEntityDump())
+        if (config.debugInventoryDump) fullObject.add("inventory", getInventoryDump())
+        if (config.debugPlayerDump) fullObject.add("player", getPlayerDump())
+
+        return fullObject
+    }
+
+
+    private fun dumpToFile(jsonObject: JsonObject) {
+        // Format the Instant to a human-readable date and time
+        // Convert epoch time to LocalDateTime
+        val dateTime = LocalDateTime.ofInstant(Instant.now(), ZoneId.systemDefault())
+
+        // Format the LocalDateTime to a human-readable date and time
+        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss")
+        val formattedDate = dateTime.format(formatter)
+
+        File("./config/partly-sane-skies/dumps/").mkdirs()
+        // Declares the file
+        val file = File("./config/partly-sane-skies/dumps/percy-dump-${formattedDate}.json")
+        file.createNewFile()
+        file.setWritable(true)
+        // Creates a new Gson object to save the data
+        // Saves teh data to the file
```

Comment:
teh

Reply:
You're going to yell at me because a comment is misspelled????????

Reply:
You want me to do a typo fix pr?